pkg/operator/sync.go

```diff
@@ -547,6 +547,32 @@ func newPodTemplateSpec(config *OperatorConfig, features map[string]bool) *corev
 	}
 }
 
+func getProxyArgs(config *OperatorConfig) []corev1.EnvVar {
+	var envVars []corev1.EnvVar
+
+	if config.Proxy != nil {
```

Comment:
Bit of a nit, but would be nicer to switch this and return early to avoid the indentation that this creates, WDYT?

Reply:
sure i'm fine with that.  I have typically tried to maintain one return point in functions for clarity but there may be a go convention here that I am not familiar with.  I can make that change.  Also, wrt injecting variable to the kube-rbac-proxy I do not think that is necessary as the kube-rbac-proxy should just be hitting the kube API.  If my understanding is incorrect let me know and I'm happy to inject the proxy in to the kube-rbac-proxy as well.

Reply:
>  have typically tried to maintain one return point in functions for clarity but there may be a go convention here that I am not familiar with.

In go the idiom tends to be to keep code as un-indented as possible, so an early return if one of your inputs is nil is quite common.

>  Also, wrt injecting variable to the kube-rbac-proxy I do not think that is necessary as the kube-rbac-proxy should just be hitting the kube API.

Yeah I think that's true. We can leave them off for now. The only thing I was wondering is whether the proxy may need to look up certificate revocation lists or something like that but I don't think go certificate libraries actually handle these. I think we are fine continuing without.